### ISSUE TITLE
Allow adding just the url to `.vsix` plugins into devfile

### ISSUE THREAD

**Issue by SDAdham** [eclipse-che/che#16266]
### Is your enhancement related to a problem? Please describe.

Currently, I have to manually add plugins via View >> Plugins in theia editor and then browse extensions from VS code and add them into theia

### Describe the solution you'd like

It would be nice if we can just simply add the list of extensions to be loaded when the workspace starts in the devfile

### Additional context

![image](https://user-images.githubusercontent.com/11182824/76091766-4a45fd00-6012-11ea-9588-8e854e33c9c7.png)

**Comment by tolusha** [comment]
But what about adding them into devfile from dashboard ?

**Comment by SDAdham** [comment]
@tolusha can you please explain more?

**Comment by tsmaeder** [comment]
@SDAdham there are a couple of points to be made here:

1. The things you see in the plugins view are not VS Code extensions, they are Che plugins
2. You can enable Che plugins in the devfile. When you edit a devfile in the dashboard, you have a tab that allows to turn plugins on/off.

**Comment by SDAdham** [comment]
@tsmaeder your message clarified:
> Use of the VS Code marketplace with non-MS products is problematic with respect to the terms of use of the marketplace. That's why we're not supporting that use case.

Closing this issue as it's no longer applicable

**Comment by SDAdham** [comment]
@tsmaeder : OK, can we at least allow uploading just `vsix` or at least provide a link to `vsix`?

Why do we really need create a replicate here https://www.eclipse.org/che/docs/che-7/using-a-visual-studio-code-extension-in-che/#publishing-a-vs-code-extension-into-the-che-plug-in-registry_using-a-visual-studio-code-extension-in-che

![image](https://user-images.githubusercontent.com/11182824/76473790-aadb9c80-644d-11ea-8697-9e2363ac7770.png)

**Comment by tsmaeder** [comment]
I agree that the restrictions on the values in the meta.yaml are due to the implementation choices of the plugin registry and I would be glad if we could get rid of that. But that's not happening in the short term, IMO.
As for referencing *.vsix files: theia runs on a container that only has what is strictly necessary to run che-theia. That's why we run plugins in their own container that has installed what they need to run. For example, the Java plugins need Java installed to run the jdt.ls language server. So in order to run a plugin, you need a suitable container to run it in.

**Comment by SDAdham** [comment]
@tsmaeder how can I obtain the right container then? Let's say I want to add https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens ? what do I do?

**Comment by ericwill** [comment]
> @tsmaeder how can I obtain the right container then? Let's say I want to add https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens ? what do I do?

Well you need to check the plugins dependencies and see if it fits any of the pre-built sidecars we are using. For example simple plugins that just need node don't need to run in a sidecar -- they can run in the main theia container.

Plugins like Java that require special dependencies are more complicated -- they need their own sidecar images in order to function. In your case we already have gitlens in the registry, so I'm assuming this question might be out of date.

**Comment by tsmaeder** [comment]
This is still a good idea, IMO
